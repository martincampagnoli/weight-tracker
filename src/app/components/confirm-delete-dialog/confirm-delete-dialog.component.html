<h2 mat-dialog-title>Delete Entry</h2>
<mat-dialog-content>
  <span>Are you sure you want to delete this entry?</span><br />
</mat-dialog-content>
<mat-dialog-actions>
  <button mat-button mat-dialog-close>No</button>
  <button mat-button [mat-dialog-close]="true" cdkFocusInitial>Yes</button>
</mat-dialog-actions>
