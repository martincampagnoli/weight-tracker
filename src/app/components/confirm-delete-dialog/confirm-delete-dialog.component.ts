import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import {
  MatDialogActions,
  MatDialogClose,
  MatDialogTitle,
  MatDialogContent,
  MatDialogRef,
} from '@angular/material/dialog';

@Component({
  selector: 'confirm-delete-dialog',
  templateUrl: 'confirm-delete-dialog.component.html',
  standalone: true,
  imports: [
    MatButtonModule,
    MatDialogActions,
    MatDialogClose,
    MatDialogTitle,
    MatDialogContent,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ConfirmDeleteDialogComponent {
  readonly dialogRef = inject(MatDialogRef<ConfirmDeleteDialogComponent>);
}
