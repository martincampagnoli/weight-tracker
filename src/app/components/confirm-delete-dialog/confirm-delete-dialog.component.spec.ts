import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialogRef } from '@angular/material/dialog';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { ConfirmDeleteDialogComponent } from './confirm-delete-dialog.component';

describe('ConfirmDeleteDialogComponent', () => {
  let component: ConfirmDeleteDialogComponent;
  let fixture: ComponentFixture<ConfirmDeleteDialogComponent>;
  let mockDialogRef: jest.Mocked<MatDialogRef<ConfirmDeleteDialogComponent>>;

  beforeEach(async () => {
    mockDialogRef = {
      close: jest.fn(),
      afterClosed: jest.fn(),
      beforeClosed: jest.fn(),
    } as any;

    await TestBed.configureTestingModule({
      imports: [ConfirmDeleteDialogComponent, NoopAnimationsModule],
      providers: [
        {
          provide: MatDialogRef,
          useValue: mockDialogRef,
        },
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(ConfirmDeleteDialogComponent);
    component = fixture.componentInstance;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Component Creation', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('Template Rendering', () => {
    it('should render dialog title', () => {
      fixture.detectChanges();

      const dialogTitle =
        fixture.debugElement.nativeElement.querySelector('[mat-dialog-title]');
      expect(dialogTitle).toBeTruthy();
    });

    it('should render dialog actions', () => {
      fixture.detectChanges();

      const dialogActions =
        fixture.debugElement.nativeElement.querySelector('mat-dialog-actions');
      expect(dialogActions).toBeTruthy();
    });

    it('should render cancel and confirm buttons', () => {
      fixture.detectChanges();

      const buttons =
        fixture.debugElement.nativeElement.querySelectorAll('button');
      expect(buttons.length).toBe(2);
    });
  });

  describe('Dialog Interactions', () => {
    it('should have No button with mat-dialog-close', () => {
      fixture.detectChanges();

      const noButton = fixture.debugElement.nativeElement.querySelector(
        'button[mat-dialog-close]'
      );
      expect(noButton).toBeTruthy();
      expect(noButton.textContent.trim()).toBe('No');
    });

    it('should have Yes button with mat-dialog-close="true"', () => {
      fixture.detectChanges();

      const buttons =
        fixture.debugElement.nativeElement.querySelectorAll('button');
      const yesButton = Array.from(buttons).find(
        (btn: any) => btn.textContent?.trim() === 'Yes'
      ) as HTMLButtonElement;
      expect(yesButton).toBeTruthy();
      expect(yesButton.textContent?.trim()).toBe('Yes');
    });
  });
});
