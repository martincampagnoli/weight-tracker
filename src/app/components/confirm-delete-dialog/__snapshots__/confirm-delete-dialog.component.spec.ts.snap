// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDeleteDialogComponent Component Creation should match snapshot 1`] = `
<confirm-delete-dialog>
  <h2
    class="mat-mdc-dialog-title mdc-dialog__title"
    id="mat-mdc-dialog-title-a1"
    mat-dialog-title=""
  >
    Delete Entry
  </h2><mat-dialog-content
    class="mat-mdc-dialog-content mdc-dialog__content"
  >
    <span>
      Are you sure you want to delete this entry?
    </span>
    <br />
  </mat-dialog-content><mat-dialog-actions
    class="mat-mdc-dialog-actions mdc-dialog__actions"
  >
    <button
      class="mdc-button mat-mdc-button-base mat-mdc-button mat-unthemed _mat-animation-noopable"
      mat-button=""
      mat-dialog-close=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-uninitialized=""
      type="button"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
        No
      </span>
      <span
        class="mat-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
    <button
      cdkfocusinitial=""
      class="mdc-button mat-mdc-button-base mat-mdc-button mat-unthemed _mat-animation-noopable"
      mat-button=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-uninitialized=""
      type="button"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
        Yes
      </span>
      <span
        class="mat-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
  </mat-dialog-actions>
</confirm-delete-dialog>
`;
